monitor/price/pricemonitoring.go

```diff
@@ -449,6 +450,21 @@ func (e *Engine) getCurrentPriceRanges(force bool) map[*bound]priceRange {
 		if !b.Active {
 			continue
 		}
+		if marketInAuction && len(e.pricesPast)+len(e.pricesNow) > 0 {
+			triggerLookback := e.now.Add(time.Duration(-b.Trigger.Horizon) * time.Second)
+			// check if trigger's not stale (newest reference price older than horizon lookback time)
+			var mostRecentObservation time.Time
+			if len(e.pricesNow) > 0 {
+				mostRecentObservation = e.now
+			} else {
+				x := e.pricesPast[len(e.pricesPast)-1]
+				mostRecentObservation = x.Time
+			}
+			if mostRecentObservation.Before(triggerLookback) {
+				b.Active = false
```

Comment:
IDK, but this looks like something that changes the spec a bit. AFAIK, the price bounds stay in place. We enter price auction, and can have it extended if the auction ends up uncrossing at a price point even further removed from the original bound that was breached. Having bounds that weren't breached expire during an auction (when there are no trades by definition) just looks a bit weird. Wouldn't it be possible for a market to be defined with an auction duration that basically ensures all bounds expire? Is that valid, or should we add validation on the governance level to make sure the price monitoring params are sensible?

Reply:
The spec was already changed a few months ago, this brings the implementation in line with it. You could have a minimum auction duration that is higher than each of the horizons, but that's perhaps desirable in some cases: effectively a way of disabling price auction extensions - you can still have a price auction though (continuous -> price auction, just not able to extend liquidity (or price) auction with a price auction)

Reply:
Ok, your suggestion from the other comment to just give price monitoring engine access to `auctionState` is probably the cleanest, will proceed on that basis.